File: back/src/main/java/br/eng/eliseu/presente/model/dto/UsuarioDto.java
package br.eng.eliseu.presente.model.dto;

import br.eng.eliseu.presente.model.PapelEnum;
import br.eng.eliseu.presente.model.StatusEnum;

import java.time.LocalDateTime;

public record UsuarioDto(
        Long id,
        String username,
        String senha,
        PapelEnum papel,
        StatusEnum status,
        LocalDateTime criadoEm,
        LocalDateTime alteradoEm,
        Long version
) {}

File: front/src/app/shared/model/dto/usuario-dto.ts
import {PapelEnum} from "@/shared/model/enum/papel.enum";
import {StatusEnum} from "@/shared/model/enum/status.enum";

export interface UsuarioDto {
    id?: number;
    username?: string;
    senha?: string; // Usado apenas para envio de nova senha
    papel?: PapelEnum;
    status?: StatusEnum;
    criadoEm?: Date;
    alteradoEm?: Date;
    version?: number;
}
